### ISSUE TITLE
warn "eats" leading unded values as if nonexistent

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#9873]
Migrated from [rt.perl.org#69126](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=69126) (status was 'resolved')

Searchable as RT69126$

**Comment by p5pRT** [comment]
### From perlbug@plan9.de

#### Created by perlbug@plan9.de



According to the perlfunc manpage\, this should print just the "at \.\.\."
line number text&#8203;:

&nbsp;   warn undef;

But instead it acts as if nothing was specified \($@&#8203; \+ line number text\)\.

In fact\, you cna specify any number of undef arguments\, and it will all
ignore them&#8203;:

&nbsp;   warn undef\, undef\, undef;    \# "$@&#8203; at\.\.\."
&nbsp;   warn undef\, undef\, undef\, 5; \# "5 at \.\.\."

One of "warn" or perlfunc is wrong\.

Die seems to suffer from the same bug\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=low

Site configuration information for perl 5.10.0:

Configured by Marc Lehmann at Sat Feb 21 02:30:27 CET 2009.

Summary of my perl5 (revision 5 version 10 subversion 0) configuration:
  Platform:
    osname=linux, osvers=2.6.24-etchnhalf.1-amd64, archname=amd64-linux
    uname='linux cerebro 2.6.24-etchnhalf.1-amd64 #1 smp mon jul 21 10:36:02 utc 2008 x86_64 gnulinux '
    config_args='-Duselargefiles -Dxxxxuse64bitint -Uuse64bitall -Dusemymalloc=n -Dcc=gcc -Dccflags=-ggdb -gdwarf-2 -g3 -Dcppflags=-DPERL_ARENA_SIZE=16368 -D_GNU_SOURCE -I/opt/include -Doptimize=-O6 -msse2 -funroll-loops -fno-strict-aliasing -Dcccdlflags=-fPIC -Dldflags=-L/opt/perl/lib -L/opt/lib -Dlibs=-ldl -lm -lcrypt -Darchname=amd64-linux -Dprefix=/opt/perl -Dprivlib=/opt/perl/lib/perl5 -Darchlib=/opt/perl/lib/perl5 -Dvendorprefix=/opt/perl -Dvendorlib=/opt/perl/lib/perl5 -Dvendorarch=/opt/perl/lib/perl5 -Dsiteprefix=/opt/perl -Dsitelib=/opt/perl/lib/perl5 -Dsitearch=/opt/perl/lib/perl5 -Dsitebin=/opt/perl/bin -Dman1dir=/opt/perl/man/man1 -Dman3dir=/opt/perl/man/man3 -Dsiteman1dir=/opt/perl/man/man1 -Dsiteman3dir=/opt/perl/man/man3 -Dman1ext=1 -Dman3ext=3 -Dpager=/usr/bin/less -Uafs -Uusesfio -Uusenm -Uuseshrplib -Dd_dosuid -Dusethreads=undef -Duse5005threads=undef -Duseithreads=undef -Dusemultiplicity=undef -Demail=perl-binary@plan9.de -Dcf_email=perl-binary@plan9.de -Dcf_by=Marc Lehmann -Dlocincpth=/opt/perl/include /opt/include -Dmyhostname=localhost -Dmultiarch=undef -Dbin=/opt/perl/bin -Dxxxusedevel -DxxxDEBUGGING -Dxxxuse_debugging_perl -Dxxxuse_debugmalloc -des'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=undef, usemultiplicity=undef
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=define, use64bitall=undef, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='gcc', ccflags ='-ggdb -gdwarf-2 -g3 -fno-strict-aliasing -pipe -I/opt/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O6 -msse2 -funroll-loops -fno-strict-aliasing',
    cppflags='-DPERL_ARENA_SIZE=16368 -D_GNU_SOURCE -I/opt/include -ggdb -gdwarf-2 -g3 -fno-strict-aliasing -pipe -I/opt/include'
    ccversion='', gccversion='4.3.2', gccosandvers=''
    intsize=4, longsize=8, ptrsize=8, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16
    ivtype='long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='gcc', ldflags ='-L/opt/perl/lib -L/opt/lib -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib /lib64 /usr/lib64
    libs=-ldl -lm -lcrypt
    perllibs=-ldl -lm -lcrypt
    libc=/lib/libc-2.7.so, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version='2.7'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E'
    cccdlflags='-fPIC', lddlflags='-shared -O6 -msse2 -funroll-loops -fno-strict-aliasing -L/opt/perl/lib -L/opt/lib -L/usr/local/lib'

Locally applied patches:
    http://public.activestate.com/cgi-bin/perlbrowse/p/34209
    http://public.activestate.com/cgi-bin/perlbrowse/p/34507
    http://www.gossamer-threads.com/lists/perl/porters/232549
    embed.fnc:Perl_vcroak NULLOK


@INC for perl 5.10.0:
    /root/src/sex
    /opt/perl/lib/perl5
    /opt/perl/lib/perl5
    /opt/perl/lib/perl5
    /opt/perl/lib/perl5
    /opt/perl/lib/perl5
    .


Environment for perl 5.10.0:
    HOME=/root
    LANG (unset)
    LANGUAGE (unset)
    LC_CTYPE=en_US.UTF-8
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/root/s2:/root/s:/opt/bin:/opt/sbin:/bin:/sbin:/usr/bin:/usr/sbin:/usr/X11/bin:/usr/games:/usr/local/bin:/usr/local/sbin:/root/pserv:.
    PERL5LIB=/root/src/sex
    PERL5_CPANPLUS_CONFIG=/root/.cpanplus/config
    PERLDB_OPTS=ornaments=0
    PERL_ANYEVENT_DBI_TESTS=1
    PERL_ANYEVENT_EDNS0=1
    PERL_ANYEVENT_NET_TESTS=1
    PERL_ANYEVENT_PROTOCOLS=ipv4,ipv6
    PERL_ANYEVENT_STRICT=1
    PERL_BADLANG (unset)
    PERL_UNICODE=E
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
@rgs - Status changed from 'new' to 'resolved'

**Comment by p5pRT** [comment]
### From @rgs

2009/9/14 perlbug@&#8203;plan9\.de \(via RT\) \<perlbug\-followup@&#8203;perl\.org>&#8203;:
> According to the perlfunc manpage\, this should print just the "at \.\.\."
> line number text&#8203;:
>
>   warn undef;
>
> But instead it acts as if nothing was specified \($@&#8203; \+ line number text\)\.
>
> In fact\, you cna specify any number of undef arguments\, and it will all
> ignore them&#8203;:
>
>   warn undef\, undef\, undef;    \# "$@&#8203; at\.\.\."
>   warn undef\, undef\, undef\, 5; \# "5 at \.\.\."
>
> One of "warn" or perlfunc is wrong\.
>
> Die seems to suffer from the same bug\.

Given what the code does\, I think it's pretty clearly a documentation
bug\, which I fixed very simply by the following commit&#8203;:

commit a96d018829db6d60ab7d14d2ba51db878e944ddd
Author&#8203;: Rafael Garcia\-Suarez \<rgarciasuarez@&#8203;gmail\.com>
Date&#8203;:   Mon Sep 14 14&#8203;:41&#8203;:10 2009 \+0200

&nbsp;    Doc nit in perlfunc for die\(\) and warn\(\) with an empty message

&nbsp;    \[perl \#69126\] warn "eats" leading unded values as if nonexistent

<details><summary>Inline Patch</summary>

```diff
diff --git a/pod/perlfunc.pod b/pod/perlfunc.pod
index 776aaf2..08c406c 100644
--- a/pod/perlfunc.pod
+++ b/pod/perlfunc.pod
@@ -1275,14 +1275,14 @@ produce, respectively

 See also exit(), warn(), and the Carp module.

-If LIST is empty and C<$@> already contains a value (typically from a
+If the output is empty and C<$@> already contains a value (typically from a
 previous eval) that value is reused after appending C<"\t...propagated">.
 This is useful for propagating exceptions:

     eval { ... };
     die unless $@ =~ /Expected exception/;

-If LIST is empty and C<$@> contains an object reference that has a
+If the output is empty and C<$@> contains an object reference that has a
 C<PROPAGATE> method, that method will be called with additional file
 and line number parameters.  The return value replaces the value in
 C<$@>.  i.e. as if C<< $@ = eval { $@->PROPAGATE(__FILE__, __LINE__) }; >>
@@ -7392,7 +7392,7 @@ Prints the value of LIST to STDERR.  If the last

```
</details>
element of LIST does
 not end in a newline\, it appends the same file/line number text as C\<die>
 does\.

\-If LIST is empty and C\<$@&#8203;> already contains a value \(typically from a
\+If the output is empty and C\<$@&#8203;> already contains a value \(typically from a
 previous eval\) that value is used after appending C\<"\\t\.\.\.caught">
 to C\<$@&#8203;>\.  This is useful for staying almost\, but not entirely similar to
 C\<die>\.
